apps/spruce/src/pages/waterfall/WaterfallMenu/GitCommitSearch.tsx

```diff
@@ -0,0 +1,80 @@
+import { useState } from "react";
+import styled from "@emotion/styled";
+import TextInput from "@leafygreen-ui/text-input";
+import { Description } from "@leafygreen-ui/typography";
+import { size } from "@evg-ui/lib/constants/tokens";
+import { useWaterfallAnalytics } from "analytics/waterfall/useWaterfallAnalytics";
+import { DropdownItem } from "components/ButtonDropdown";
+import { ConfirmationModal } from "components/ConfirmationModal";
+import { useQueryParams } from "hooks/useQueryParam";
+import { WaterfallFilterOptions } from "../types";
+
+interface GitCommitSearchProps {
+  setMenuOpen: (open: boolean) => void;
+}
+
+export const GitCommitSearch: React.FC<GitCommitSearchProps> = ({
+  setMenuOpen,
+}) => {
+  const { sendEvent } = useWaterfallAnalytics();
+  const [, setQueryParams] = useQueryParams();
+
+  const [modalOpen, setModalOpen] = useState(false);
+  const [commitHash, setCommitHash] = useState("");
+
+  const onCancel = () => {
+    setModalOpen(false);
+    setMenuOpen(false);
+  };
+
+  const onConfirm = () => {
+    sendEvent({ name: "Filtered by git commit" });
+    setQueryParams({
+      [WaterfallFilterOptions.Revision]: commitHash,
+    });
+    onCancel();
+  };
+
+  return (
+    <>
+      <DropdownItem
+        data-cy="git-commit-search"
+        onClick={() => {
+          setModalOpen(true);
+          sendEvent({ name: "Viewed git commit modal" });
+        }}
+      >
+        Search by Git Hash
+      </DropdownItem>
+      <ConfirmationModal
+        cancelButtonProps={{
+          children: "Cancel",
+          onClick: onCancel,
+        }}
+        confirmButtonProps={{
+          children: "Submit",
+          onClick: onConfirm,
+          disabled: commitHash.trim().length < 7,
+        }}
+        data-cy="git-commit-search-modal"
+        open={modalOpen}
+        title="Search by Git Commit Hash"
+      >
+        <StyledDescription>
+          Note: This is an experimental feature that works best without any
+          filters. Searching for a git hash will clear all applied filters.
```

Comment:
IMO this isn't really an experimental feature anymore since it's been used successfully for at least a year. I also don't think filters have any impact on the search now that filtering is client-side.
```suggestion
          Searching for a git hash will clear all applied filters.
```